api/src/main/java/org/opennms/integration/api/v1/timeseries/TimeSeriesStorage.java

```diff
@@ -42,8 +42,11 @@
     /** Stores a list of Samples in the timeseries database. */
     void store(List<Sample> samples) throws StorageException;
 
-    /** Returns all metrics which are stored in the time series database which contain all given tags (as intrinsic tags or meta tags). */
-    List<Metric> getMetrics(Collection<Tag> tags) throws StorageException;
+    /**
+     * Returns all metrics which are stored in the time series database which pass all TagMatcher.
+     * In order to "pass" a Metric must have at least one matching searchable tag for each TagMatcher.
```

Comment:
What is the expected behavior if the collection is empty? I would say we throw some exception, rather than returning *all* metrics.

Reply:
:+1: I added a javadoc comment and enforce it by our AbstractIntegrationTest